File: backend/src/main/java/com/programmers/pcquotation/global/Properties/OAuth2RegistrationProperties.java
package com.programmers.pcquotation.global.Properties;

import java.util.List;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "spring.security.oauth2.client.registration")
public record OAuth2RegistrationProperties(
	RegistrationConfig kakao,
	RegistrationConfig naver
) {
	public record RegistrationConfig(
		String clientId,
		String clientSecret,
		String redirectUri,
		String authorizationGrantType,
		String clientAuthenticationMethod,
		String clientName,
		List<String> scope
	) {
	}
}

File: backend/src/main/java/com/programmers/pcquotation/global/properties/OAuth2RegistrationProperties.kt
package com.programmers.pcquotation.global.properties

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "spring.security.oauth2.client.registration")
data class OAuth2RegistrationProperties(
    val kakao: RegistrationConfig,
    val naver: RegistrationConfig
) {
    data class RegistrationConfig(
        val clientId: String,
        val clientSecret: String,
        val redirectUri: String,
        val authorizationGrantType: String,
        val clientAuthenticationMethod: String,
        val clientName: String,
        val scope: List<String>? = null
    )
}